### PR TITLE
Batch produce to a partition

### LINKED ISSUE #37: Batch produce to a partition
There's currently no way to batch produce to a specific partition, something like:

* `topic/partition/message_list`
```elixir
Kaffe.Producer.produce_sync("topic", 2, [{"key1", "value1"}, {"key2", "value2"}])
```

I can work on this and send a PR, is this something that you would merge back into `kaffe`? Or is there a reason to not having this feature?

Thanks

### PR DESCRIPTION
**Pull request by rodrigosaito**
This fixes #37

### CODE REVIEW

File: lib/kaffe/producer.ex [hunk 1]
```diff
@@ -83,6 +97,11 @@ defmodule Kaffe.Producer do
   ## internal
   ## -------------------------------------------------------------------------
 
+  defp produce_list(topic, partition, message_list) when is_list(message_list) do
+    Logger.debug "event#produce_list topic=#{topic} partition=#{partition}"
+    @kafka.produce_sync(client_name(), topic, partition, "ignored", message_list)
```
Comment (rwdaigle) [inline lib/kaffe/producer.ex]: Is there any value in having this function utilize `produce_list_to_topic` (as the existing `produce_list` does) so there aren't two independent paths to the brod client invocation? I realize there might be some finagling required to support a consistent format between producing to a static partition vs. producing to a dynamically chosen partition, but it seems like we should minimize the branching in that last mile logic.

Thoughts?

Comment (rodrigosaito) [inline lib/kaffe/producer.ex]: @rwdaigle I wondered the same thing when I was writing the code,  I will try to do that...

**Comment by rodrigosaito** [comment]
@rwdaigle updated PR, see what you think...

**Comment by objectuser** [comment]
@rodrigosaito Looking good!

I get some warnings when I compile:

```
$ mix test
Compiling 18 files (.ex)
warning: clauses for the same def should be grouped together, def produce_sync/2 was previously defined (lib/kaffe/producer.ex:44)
  lib/kaffe/producer.ex:73

warning: clauses for the same def should be grouped together, def produce_sync/3 was previously defined (lib/kaffe/producer.ex:58)
  lib/kaffe/producer.ex:83
```

Would you fix (and rebase from `master`)?

**Comment by rodrigosaito** [comment]
Sure... will do that tomorrow or day after tomorrow.

**Comment by rodrigosaito** [comment]
@objectuser done!

**Pull request merged**